clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AbstractAtLeastOnceKafkaConsumer.java

```diff
@@ -209,17 +211,16 @@ private AbstractAtLeastOnceKafkaConsumer(final KafkaConsumer<String, Buffer> kaf
     }
 
     /**
-     * Closes the underlying Kafka consumer.
+     * Closes the underlying Kafka consumer and tries to commit the current offsets before.
      * <p>
      * This does not invoke the close handler.
      *
-     * @return A future that will complete when the Kafka consumer is closed.
+     * @return A future that will complete when the Kafka consumer is closed. If the offsets have been committed
+     *         successfully, the future will succeed, otherwise, it will fail.
      */
     @Override
     public Future<Void> stop() {
-        final Promise<Void> promise = Promise.promise();
-        kafkaConsumer.close(promise);
-        return promise.future();
+        return tryCommitAndClose();
```

Comment:
When testing this, I always seem to be getting an exception when stopping the consumer:
````
# consumer.stop() called here
12:53:45.061 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.k.c.AbstractAtLeastOnceKafkaConsumer - committing the current offsets
12:53:45.425 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.k.c.AbstractAtLeastOnceKafkaConsumer - Polled 0 records
12:53:45.445 [vert.x-eventloop-thread-0] ERROR o.e.h.c.k.c.AbstractAtLeastOnceKafkaConsumer - Error committing offsets: org.apache.kafka.common.errors.WakeupException
12:53:45.458 [vert.x-kafka-consumer-thread-0] INFO  o.a.k.c.c.i.ConsumerCoordinator - [Consumer clientId=test-a5bee52f-ee67-4381-b682-7c62b95bad76, groupId=myGroup] Revoke previously assigned partitions hono.command_internal.myAdapterInstanceId-0
12:53:45.459 [vert.x-kafka-consumer-thread-0] INFO  o.a.k.c.c.i.AbstractCoordinator - [Consumer clientId=test-a5bee52f-ee67-4381-b682-7c62b95bad76, groupId=myGroup] Member test-a5bee52f-ee67-4381-b682-7c62b95bad76-d9a878d8-899f-4819-8ca5-fe6b13e018f3 sending LeaveGroup request to coordinator localhost:49186 (id: 2147483646 rack: null) due to the consumer is being closed
12:53:45.462 [vert.x-eventloop-thread-0] ERROR o.e.h.c.k.c.AbstractAtLeastOnceKafkaConsumer - Closing consumer with cause
org.eclipse.hono.client.kafka.consumer.KafkaConsumerCommitException: org.apache.kafka.common.errors.WakeupException
	at org.eclipse.hono.client.kafka.consumer.AbstractAtLeastOnceKafkaConsumer.lambda$commit$6(AbstractAtLeastOnceKafkaConsumer.java:290)
	at io.vertx.core.Future.lambda$recover$6(Future.java:499)
	at io.vertx.core.impl.FutureImpl$Handlers.handle(FutureImpl.java:228)
	at io.vertx.core.impl.FutureImpl$Handlers.handle(FutureImpl.java:224)
	at io.vertx.core.impl.FutureImpl.tryFail(FutureImpl.java:195)
	at io.vertx.core.impl.FutureImpl.fail(FutureImpl.java:125)
	at io.vertx.core.impl.FutureImpl.handle(FutureImpl.java:178)
	at io.vertx.core.impl.FutureImpl.handle(FutureImpl.java:21)
	at io.vertx.kafka.client.consumer.impl.KafkaConsumerImpl.lambda$commit$5(KafkaConsumerImpl.java:366)
	at io.vertx.kafka.client.consumer.impl.KafkaReadStreamImpl.lambda$null$1(KafkaReadStreamImpl.java:123)
````